bundles/com.zeligsoft.domain.dds4ccm/src/com/zeligsoft/domain/dds4ccm/utils/DDS4CCMMigrationModelTypeUtil.java

```diff
@@ -69,35 +92,268 @@ private static boolean migrateAll(Model model, boolean dryrun) {
+		}
+	}
+	
+	/**
+	 * Retrieves 'ConnectorDef' object for the default value of connectorType set in the preference page
+	 * 
+	 * @param defaultConnectorType
+	 *          Preferred value of 'ConnectorType' for a sync/async port for being used during migration 
+	 * @param model
+	 * 			an atcd model that is being migrated           
+	 * @return a 'ConnectorDef' object that corresponds to the default value of the 'ConnectorType' 
+	 *         
+	 */  
+	private static ConnectorDef getDefaultConnectorDef(String defaultConnectorType, Model model){
+		
+		String pathmapConnectorLib = getPathmapConnectorLib(defaultConnectorType);
+		URI uriConnectorLib = URI.createURI(pathmapConnectorLib);
+		Package packageConnectorLib = getImportedConnectorPackage(model, uriConnectorLib);
+		ConnectorDef connectorDef = getConnectorDef(packageConnectorLib);
+		
+		return connectorDef;
+	}
+	
+	/**
+	 * Retrieves pathmap of a connector library
+	 * 
+	 * @param preferredConnectorType
+	 *          Preferred value of 'ConnectorType' to be used during migration 
+	 *            
+	 * @return pathmap of a connector library which corresponds to the input connectorType value  
+	 *         
+	 */
+	
+	private static String getPathmapConnectorLib(String preferredConnectorType){
+		
+		String pathmapConnectorLib = "";
+		
+		String pathmapAMIConnector = "pathmap://DDS4CCM_AMI_LIBRARIES/CCM_AMI.uml";
+		String pathmapCORBAConnector = "pathmap://DDS4CCM_CORBA_LIBRARIES/CCM_CORBA.uml";
+		
+		if(preferredConnectorType.equals(ConnectorType.AMI4CCM_Connector.name())){
+			pathmapConnectorLib = pathmapAMIConnector; 
+		}else if(preferredConnectorType.equals(ConnectorType.CORBA4CCM_Connector.name())){
+			pathmapConnectorLib = pathmapCORBAConnector;
+		}
+		
+		return pathmapConnectorLib;
+	}
+	
+	/**
+	 * Import a package represented by the input 'packageUri' in the input 'model' and return it. 
+	 * If the package is already imported, return the imported package.  
+	 * 
+	 * @param model
+	 *          model object where the package should be imported
+	 * @param packageUri
+	 * 			URI of the package object that is being imported          
+	 *            
+	 * @return the imported package object  
+	 *         
+	 */
+	private static Package getImportedConnectorPackage(Model model, URI packageUri){
+		
+		Package importedConnectorPackage = null;
+		boolean isAlreadyImported = false;
+	
+		List<PackageImport> listPackageImport = model.getPackageImports();
+		
+		for(PackageImport pi: listPackageImport){
+			
+			if(pi.getImportedPackage().eResource().getURI().equals(packageUri)){
+				isAlreadyImported = true;
+				importedConnectorPackage = pi.getImportedPackage();
+				break;
+			}
+		}
+		
+		if(!isAlreadyImported){
+			org.eclipse.emf.ecore.resource.Resource connectorPackageResource = model.eResource().getResourceSet().createResource(packageUri);			
+			
+			try {
+				connectorPackageResource.load(null);
```

Comment:
This does not work correctly. After I migrate a model, I do not see the package imports in the 'Details' tab of the model editor (in the Model Libraries box). The following code has worked correctly in other circumstances:

final org.eclipse.uml2.uml.Package library = (Package) UMLModeler.openModelResource(uri);
final PackageImport pi = model.createPackageImport(library);

Addendum: If I close and re-open the model/model-editor, I do see the package imports. Still, the logic here seems to be needlessly complex.